Add 1.8.0 release blogpost (#195)
kubewarden/kubewarden.io: content/blog/2023/10/kubewarden-1.8-release.md

---
title: Kubewarden 1.8.0 release
authors:
  - Víctor Cuadrado
date: 2023-10-11
---

Today we are happy to announce the release of Kubewarden 1.8.0! 🎉🥳

This is a small release, focused on [OpenTelemetry](https://opentelemetry.io/).

The OpenTelemetry Protocol (OTLP) got its [first 1.0.0
version](https://github.com/open-telemetry/opentelemetry-proto/releases/tag/v1.0.0)

> in July 2023, and the OpenTelemetry stack is becoming more stable; several

Reviewer: ```suggestion
in July 2023; several
```

It seems a bit contradictory/confusing to read on one line that otel is becoming more stable, but then on the next paragraph to read the entire opposite. So either we use a synonym, or we just drop that like in the suggestion from above.

libraries got their first 1.0.0 release, such as the [Go metric
SDK](https://opentelemetry.io/blog/2023/otel-go-metrics-sdk-stable/) or the
[.NET Automatic
Instrumentation](https://opentelemetry.io/blog/2023/otel-dotnet-auto-instrumentation/).

> Still, the OTLP stack is not yet stable, and unannounced backwards-incompatible

Reviewer: ```suggestion
Still, the OpenTelemetry stack is not yet stable, and unannounced backwards-incompatible
```

I prefer to avoid acronyms that are probably not known by the average reader

changes still happen. You can have a look at the status of each of their
libraries and protocols [here](https://opentelemetry.io/status).

For Kubewarden, this means we should expect more users to configure and fine
tune the deployment to fit their stacks.

## Changes

With 1.8.0, the Kubewarden Controller now separately reconciles the configuration of both

> tracing and metrics for PolicyServers. In the past, enabling metrics in

Reviewer: ```suggestion
tracing and metrics for Policy Server instances. In the past, enabling metrics in
```

Kubewarden meant enabling both, and now users can decide to enable each separately, as we

> have realized from the requests raised in GitHub.

Reviewer 2: maybe `in GitHub` can be rephrased with `from the community`

Author: Better! done in https://github.com/kubewarden/kubewarden.io/pull/195/commits/da56fe45efb8b26278c20dfd6626d670a90559d9.

We have made the concious decision to make this a breaking change in the helm

> chart `values.yaml`, which we never do.

Reviewer 2: maybe we could replace `we never do` with something like `it is uncommon`

Reviewer 3: Yeah, 'we never do' doesn't work as are doing it. How about 'which we rarely do'?

Author: Better! done in https://github.com/kubewarden/kubewarden.io/pull/195/commits/da56fe45efb8b26278c20dfd6626d670a90559d9.

We think it simplifies the adoption of OTLP changes in the future.

The change is the following:

```diff
# values.yaml
# open-telemetry options
telemetry:
- enabled: False
  metrics:
+   enabled: False
    port: 8080
  tracing:
+   enabled: False
    jaeger: {}
    # OTLP/Jaeger endpoint to send traces to
    # endpoint: "all-in-one-collector.jaeger.svc.cluster.local:4317"
```

#### Example

Up until 1.7.x inclusive we had:

```yaml
---
# values.yaml
# Up until 1.7.x inclusive
telemetry:
  enabled: True
  metrics:
    port: 8080
  tracing:
    jaeger:
      endpoint: "my-open-telemetry-collector.jaeger.svc.cluster.local:14250"
      tls:
        insecure: true
```

Starting with 1.8.0 we have:

```yaml
---
# values.yaml
# Starting with 1.8.0:
telemetry:
  metrics:
    enabled: True
    port: 8080
  tracing:
    enabled: True
    jaeger:
      endpoint: "my-open-telemetry-collector.jaeger.svc.cluster.local:4317"
      tls:
        insecure: true
```

> ## Configuring OTLP manually

Reviewer: ```suggestion
## Configuring OpenTelemetry manually
```

As always, users can decide instead to configure metrics and/or tracing support on

> their own. For example in case that they want only some of the Kubewarden

Reviewer 2: Users may still configure metrics and/or tracing support themselves, as desired. 
or
Users have the option to manually configure metrics and/or tracing support.

Author: Better! done in https://github.com/kubewarden/kubewarden.io/pull/195/commits/da56fe45efb8b26278c20dfd6626d670a90559d9.

Still, the OpenTelemetry stack is not yet stable, and unannounced backwards-incompatible
changes still happen. You can have a look at the status of each of their
libraries and protocols [here](https://opentelemetry.io/status).

For Kubewarden, this means we should expect more users to configure and fine
tune the deployment to fit their stacks.

## Changes

With 1.8.0, the Kubewarden Controller now separately reconciles the configuration of both
tracing and metrics for Policy Server instances. In the past, enabling metrics in
Kubewarden meant enabling both, and now users can decide to enable each separately, as we
have realized from the requests raised by the community.

We have made the concious decision to make this a breaking change in the helm
chart `values.yaml`, which is uncommon.
We think it simplifies the adoption of OTLP changes in the future.

In addition, since `v0.85.0` OpenTelemetry Collector no longer includes
exporters for the native Jaeger format. This change was
prompted because Jaeger has support for OTLP. See the [announcement
here](https://opentelemetry.io/blog/2023/jaeger-exporter-collector-migration).

> Hence we now take care of setting `exporters.otlp/jaeger` instead of
> `exporters.jaeger` the OpenTelemetryCollector CR we create. For the user, this

Reviewer: ```suggestion
Now the OpenTelemetryCollector resource created by the helm chart is configured
to use  `exporters.otlp/jaeger` instead of `exporters.jaeger`. For the user, this
```
